### PR TITLE
Add content compression support to StyxServer object

### PR DESCRIPTION
**Pull request by chrisgresty**
See issue #593

### CODE REVIEW

**Comment by chrisgresty** [comment]
> I see some of these tests might be equivalent to the ones in CompressionSpec. Do you think we need both?

There are now two different paths to the same compression logic - one via the current StyxServer, and one via the new StyxHttpServer. So both need testing. I don't think it's enough to just test that the correct config is set, the test should prove that the request gets through the server to the compression logic as well. Hence duplicating the tests (more or less).

Then it becomes a question of where. Perhaps it would have been better to extend the CompressionSpec test, but the existing tests for StyxHttpServer were already executing at the kind of level that are needed for this test, so it seemed to me that the expected place to find these tests would be there.

File: components/proxy/src/main/kotlin/com/hotels/styx/servers/StyxHttpServer.kt [hunk 1]
```diff
@@ -79,6 +80,7 @@ private data class StyxHttpServerTlsSettings(
 private data class StyxHttpServerConfiguration(
         val port: Int,
         val handler: String,
+        val compressResponses: Boolean,
```
Comment (mikkokar) [inline components/proxy/src/main/kotlin/com/hotels/styx/servers/StyxHttpServer.kt]: I believe this defaults to `false`?

File: components/proxy/src/test/kotlin/com/hotels/styx/servers/StyxHttpServerTest.kt [hunk 2]
```diff
@@ -302,12 +344,25 @@ private fun createConnection(port: Int) = NettyConnectionFactory.Builder()
 
 private val response = HttpResponse.response(OK)
         .header("source", "secure")
+        .header("content-type", "text/plain")
         .body("Hello, test!", UTF_8)
         .build()
 
+private val compressedResponse = HttpResponse.response(OK)
```
Comment (mikkokar) [inline components/proxy/src/test/kotlin/com/hotels/styx/servers/StyxHttpServerTest.kt]: Static import `response`.

**Comment by mikkokar** [comment]
> I see some of these tests might be equivalent to the ones in CompressionSpec. Do you think we need both?

At the moment, yes. Because there are two separate ways to start servers. But we will, sooner or later (hope sooner), to eliminate the old way.

**Pull request merged**